src/meta/src/hummock/manager/compaction.rs

```diff
@@ -102,4 +116,149 @@ impl HummockManager {
         let ctx = dynamic_level_core.get_priority_levels(&levels, &status.level_handlers);
         ctx.score_levels
     }
+
+    pub async fn handle_pull_task_event(
+        &self,
+        context_id: u32,
+        pull_task_count: usize,
+        compaction_selectors: &mut HashMap<TaskType, Box<dyn CompactionSelector>>,
+    ) {
+        assert_ne!(0, pull_task_count);
+        if let Some(compactor) = self.compactor_manager.get_compactor(context_id) {
+            let (groups, task_type) = self.auto_pick_compaction_groups_and_type().await;
+            if !groups.is_empty() {
+                let selector: &mut Box<dyn CompactionSelector> =
+                    compaction_selectors.get_mut(&task_type).unwrap();
+
+                let mut generated_task_count = 0;
+                let mut existed_groups = groups.clone();
+                let mut meet_error = false;
+                let mut wait_compact_groups: HashSet<CompactionGroupId> = HashSet::default();
+
+                while generated_task_count < pull_task_count && !meet_error {
+                    let compact_ret = self
+                        .get_compact_tasks(std::mem::take(&mut existed_groups), selector)
+                        .await;
+
+                    match compact_ret {
+                        Ok((compact_tasks, trivial_tasks)) => {
+                            if compact_tasks.is_empty() {
+                                break;
+                            }
+                            generated_task_count += compact_tasks.len();
+                            for task in trivial_tasks {
+                                existed_groups.push(task.compaction_group_id);
+                                wait_compact_groups.insert(task.compaction_group_id);
+                            }
+                            for task in compact_tasks {
```

Comment:
It is possible to send more tasks than `pull_task_count` to compactor because `get_compact_tasks` will try to get tasks from all compaction groups, which violate the contract between meta and compactor. We should pass the expected number of tasks to be generated to `get_compact_tasks` and assert that the returned number of `compacto_tasks` is not larger thant the expected number of tasks.